qiskit/pulse/logical_elements_frames.py

```diff
@@ -0,0 +1,499 @@
+        return self._index
+
+    def _validate_index(self, index) -> None:
+        """Raise a ``PulseError`` if the qubit index is invalid. Namely, check if the index is a
+        non-negative integer.
+
+        Raises:
+            PulseError: If ``identifier`` (index) is a negative integer.
+        """
+        pass
+        if not isinstance(index, (int, np.integer)) or index < 0:
+            raise PulseError("Qubit index must be a non-negative integer")
+
+
+class MeasurementFrame(Frame):
+    """A frame associated with the measurement of a qubit.
+
+    ``MeasurementFrame`` is a frame associated with the measurement of a specific qubit.
+    If not set otherwise, the initial frequency of the frame will be taken as the default
+    measurement frequency provided by the backend during compilation.
+    """
+
+    def __init__(self, qubit_index: int):
+        """Create ``MeasurementFrame``.
+
+        Args:
+            qubit_index: The index of the qubit represented by the frame.
+        """
+        self._validate_index(qubit_index)
+        self._index = qubit_index
+        super().__init__(self.name)
+
+    @property
+    def qubit_index(self) -> int:
+        """Return the qubit index of the measurement frame."""
+        return self._index
+
+    @property
+    def name(self) -> str:
+        """Return the name of the frame."""
+        return f"MeasurementFrame{self.qubit_index}"
+
+    def _validate_index(self, index) -> None:
+        """Raise a ``PulseError`` if the qubit index is invalid. Namely, check if the index is a
+        non-negative integer.
+
+        Raises:
+            PulseError: If ``index`` is a negative integer.
+        """
+        pass
+        if not isinstance(index, (int, np.integer)) or index < 0:
+            raise PulseError("Qubit index must be a non-negative integer")
+
+
+class MixedFrame:
+    """Representation of a ``LogicalElement`` and ``Frame`` combination.
+
+    Every pulse needs to be associated with both a ``LogicalElement`` and a ``Frame``. The combination
+    of the two is called a mixed frame, and is represented by a ``MixedFrame`` object. The ``MixedFrame``
+    is closely related to the legacy ``Channel``, but provides a more flexible framework.
+
+    In most cases the ``MixedFrame`` is used more by the compiler, and a pulse program can be written
+    without ``MixedFrame``s, by setting ``LogicalElement`` and ``Frame`` independently. However,
+    in some cases using ``MixedFrame``s can better convey the meaning of the code, and change the
+    compilation process. One example are shift/set frequency/phase instructions which are not
+    broadcasted to other ``MixedFrame``s if applied on a specific ``MixedFrame`` (unlike the behavior
+    of ``Frame``). A second example is the ``CRMixedFrame`` which indicates to the compiler what is the
+    purpose of the pulses played on the mixed frame, and allows for extra validation - namely, validating
+    that qubits are coupled.
```

Comment:
```suggestion
    of ``Frame``). User can also subclass ``MixedFrame` for a particular combination of
    logical elements and frames as if a syntactic sugar. This might increase the readability
    of user pulse program. Such example may include cross resonance architecture,
    in which a pulse is played on a target qubit frame and applied to a control qubit
    logical element.
```
Please do not mention particular subclass.